api/api/management/commands/updatedb.py

```diff
@@ -37,67 +46,131 @@ def handle(self, *args: Any, **options: Any):
             print("Utilize o comando 'updatedb -h' para mais informações.")
             return
 
-        # Obtem o ano e o período anterior ao período atual
+        # Obtém o ano e o período anterior ao período atual
         previous_period_year, previous_period = sessions.get_previous_period()
 
         # Apaga as disciplinas do período anterior
-        delete_all_departments_using_year_and_period(
+        dbh.delete_all_departments_using_year_and_period(
             year=previous_period_year, period=previous_period)
 
         if options["delete"]:
             for year, period in choices:
-                self.delete_period(year=year, period=period)
+                thread = threading.Thread(
+                    target=self.delete_period, args=(year, period,))
+                threads.append(thread)
+                thread.start()
+            
+            for thread in threads:
+                thread.join()
+            threads.clear()
+
             return
 
-        departments_ids = web_scraping.get_list_of_departments()
+        departments_ids = get_list_of_departments()
 
         if departments_ids is None:
             self.display_error_message("department_ids")
             return
 
         print("Atualizando o banco de dados...")
 
-        for year, period in choices:
+        def start_update_year_period(year: str, period: str):
             try:
                 start_time = time()
-                self.update_departments(
-                    departments_ids=departments_ids, year=year, period=period)
+                print(f"Começando atualização de {year}/{period}")
+                with transaction.atomic():
```

Comment:
Para o `transaction.atomic()` ter efetividade, não seria necessário adicionar nas configurações o `ATOMIC_REQUESTS = True`? 

Fonte: [Tying transactions to HTTP requests](https://docs.djangoproject.com/en/5.0/topics/db/transactions/#tying-transactions-to-http-requests)

Reply:
Na verdade a documentação do Django ficou um pouco confusa mesmo, ela diz que a utilização do `ATOMIC_REQUESTS = True` faz com que todas as funções de view fiquem atómicas:

> In practice, this feature wraps every view function in the [atomic()](https://docs.djangoproject.com/en/5.0/topics/db/transactions/#django.db.transaction.atomic) decorator described below.

Para utilizar a atomicidade somente em pontos específicos podemos fazer só o uso do `transaction.atomic()` como descrito em: 

> Atomicity is the defining property of database transactions. atomic allows us to create a block of code within which the atomicity on the database is guaranteed.

Fonte: [Controlling transactions explicitly](https://docs.djangoproject.com/en/5.0/topics/db/transactions/#controlling-transactions-explicitly)